scoreboard/webutils.py

```diff
@@ -155,22 +155,41 @@ def format_trow(game: orm.Game, index: int,
     
     if datatables:
         t += """<script>
-             $(document).ready(function(){{
-                 $('#{id}').DataTable({{
-                     "columnDefs": [
-                         {{ "searchable": false, "targets": [0,1,5,6,7,9] }},
-                         {{ "orderable": false, "targets": [7,9] }}
-                     ],
-                     "order": [[0, "asc"]],
-                     "info": false,
-                     "lengthChange": false,
-                     "oLanguage": {{
-                         "sSearch": "Filter:"
-                     }},
-                     "pagingType": "numbers"
-                 }});
-             }});
-             </script>"""
+                $(document).ready(function(){{
+                    $('#{id}').DataTable({{
+                        "columnDefs": [
+                            {{ "searchable": false, "targets": [0,1,5,6,7,9] }},
+                            {{ "orderable": false, "targets": [7,9] }}
+                        ],
+                        "order": [[0, "asc"]],
+                        "info": false,
+                        "lengthChange": false,
+                        "oLanguage": {{
+                            "sSearch": "Filter:"
+                        }},
+                        "pagingType": "numbers"
+                    }});
+                    
+                    $('#{id}_wrapper input[type=search]')
+                        .off('cut input keypress keyup paste search')
+                        .on( 'keyup change', function () {{
+                        //console.log("searching for: "+this.value);
+                        var tokens = this.value.trim().split(/\s+/);
+                        // modify tokens to be \bXXXX, if two letters also let match \b..XX\b
+                        tokens = tokens.map(function(t) {{
+                            t = "\\\\b" + t + (t.length==2 ? "|\\\\b.."+t+"\\\\b" : "");
+                            return t;
+                        }});
+                        //console.log(tokens);
+                        // AND tokens together
+                        var regex = "(?=" + tokens.join(")(?=") + ")";
+                        //console.log(regex);
+                        table = $('#{id}').DataTable();
+                        table.search(regex, true, false);
+                        table.draw();
```

Comment:
Just checking -- is this necessary? I don't know datatables at all, but it feels unneeded